core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -1156,6 +1156,9 @@ class LogManager(logDirs: Seq[File],
       // we will add metrics back after sourceLog remove the metrics
       destLog.removeLogMetrics()
       destLog.updateHighWatermark(sourceLog.highWatermark)
+      sourceLog.leaderEpochCache.foreach(cache =>
+        cache.latestEntry.map(entry =>
+          destLog.maybeAssignEpochStartOffset(entry.epoch, entry.startOffset)))
```

Comment:
Have a doubt. When becoming a new leader, `maybeAssignEpochStartOffset` has been executed in `makeLeader`, and the latest epoch and the LEO of the new leader have been assigned. Why do we need to copy it again here?

Reply:
If not, there wouldn't be leader epoch cache for the log after altering log dir without any produce traffic. As written in description, "The cache for each partition is updated only when appending new messages or becoming leader. If there is no traffic and the replica is already the leader, the cache will not be updated any more."